plugin/vimteractive.vim

```diff
@@ -12,96 +12,60 @@
 " Name of the vimteractive terminal buffer
 let g:vimteractive_buffer_name = "vimteractive_buffer"
 let g:vimteractive_terminal = ''
+let g:current_term_type = ''
 
 " Variables for running the various sessions
-let g:vimteractive_ipython_command = 'ipython --matplotlib --no-autoindent'
-let g:vimteractive_python_command = 'python'
-let g:vimteractive_bash_command = 'bash'
-let g:vimteractive_zsh_command = 'zsh'
-let g:vimteractive_julia_command = 'julia'
-let g:vimteractive_maple_command = 'maple -c "interface(errorcursor=false);"'
-
-" User commands
-" =============
-
-" Commands to start a session
-command!  Iipython :call Vimteractive_session(g:vimteractive_ipython_command) 
-command!  Ipython  :call Vimteractive_session(g:vimteractive_python_command)  
-command!  Ibash    :call Vimteractive_session(g:vimteractive_bash_command)
-command!  Izsh     :call Vimteractive_session(g:vimteractive_zsh_command)
-command!  Ijulia   :call Vimteractive_session(g:vimteractive_julia_command)
-command!  Imaple   :call Vimteractive_session(g:vimteractive_maple_command)
+"
+if !has_key(g:, 'vimteractive_commands')
+	let g:vimteractive_commands = { }
+endif
+
+let g:vimteractive_commands.ipython = 'ipython --matplotlib --no-autoindent'
+let g:vimteractive_commands.python = 'python'
+let g:vimteractive_commands.bash = 'bash'
+let g:vimteractive_commands.zsh = 'zsh'
+let g:vimteractive_commands.julia = 'julia'
+let g:vimteractive_commands.maple = 'maple -c "interface(errorcursor=false);"'
+let g:vimteractive_commands.clojure = 'clojure'
+
+" Override default shells for different filetypes
+if !has_key(g:, 'vimteractive_default_shells')
```

Comment:
Personally I'd quite like the default python shell to be the same one as g:vimteractive_commands.ipython (as the normal python REPL is pretty basic), however I could just as easily add this to my .vimrc. What do you think?

Reply:
I don't like the idea of making IPython default Python shell. First of all, it can simply be not available. We can somehow check that IPython is present (e. g. with `ipython --version`), but I suppose that if someone wants to use IPython as default , they can just change `.vimrc`.
There is a more interesting question: should default Python command be `python` or `python3`? And should we provide `:Ipython2` and `Ipython3` (and `:Iipython2`, `:Iipython3`) commands?

Reply:
>I don't like the idea of making IPython default Python shell. First of all, it can simply be not available. We can somehow check that IPython is present (e. g. with ipython --version), but I suppose that if someone wants to use IPython as default , they can just change .vimrc.

Agreed.

>There is a more interesting question: should default Python command be python or python3? And should we provide :Ipython2 and Ipython3 (and :Iipython2, :Iipython3) commands?

￼Given the pending [deprecation of python 2](https://pythonclock.org/), it's probably best to leave it as it is at the moment. Your update that now allows users to set what the default command is for a given filetype is definitely the best solution for next year, when I imagine there will be a long period of the meaning of `python` changing between systems

Reply:
The problem here is in [https://www.python.org/dev/peps/pep-0394/](PEP 394), that enforces usage of Python 2 as `python`. `python` executable will always point to Python 2 and it will be less and less useful over time.

Reply:
I guess it's wishful thinking on my part as an Arch user that "[Future changes to this recommendation](https://www.python.org/dev/peps/pep-0394/#future-changes-to-this-recommendation)"  will come soon after python2's deprecation. I've just pushed a small update that adds python2 and python3 in addition to python in the spirit of [PEP 394](https://www.python.org/dev/peps/pep-0394/)

Reply:
I hope they'll change PEP 394, but it will take a long time before distros, especially Debian-based ones, will adopt new behavior.

Yeah, that update is fine. I don't use IPython, but consider doing so for IPython as well.